src/main/java/org/ow2/proactive/catalog/util/ArchiveManagerHelper.java

```diff
@@ -149,24 +159,38 @@ public ZipArchiveContent compressZIP(List<CatalogObjectRevisionEntity> catalogOb
         }
 
         try (ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(byteArrayArchive)) {
-            ZipUtil.iterate(byteArrayInputStream, (in, zipEntry) -> filesList.add(process(in, zipEntry)));
+            ZipUtil.iterate(byteArrayInputStream, (in, zipEntry) -> checkAndAddFileFromZip(filesList, in, zipEntry));
         } catch (IOException ioe) {
             throw new RuntimeException(ioe);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)